Core/World/Entities/Players/Player.cs

```diff
@@ -912,6 +917,96 @@ private void ChangePlayerWeaponSlot(WeaponSlot slot)
         }
     }
 
+    private static int GetWeaponGroupIndex(TickCommands tickCommand) =>
+        (int)tickCommand - (int)TickCommands.WeaponGroup1;
+    
+    private static readonly TickCommands[] WeaponGroupCommands = new TickCommands[]
+    {
+        TickCommands.WeaponGroup1,
+        TickCommands.WeaponGroup2,
+        TickCommands.WeaponGroup3,
+        TickCommands.WeaponGroup4
+    };
+
+    private static TickCommands GetWeaponGroupCommand(TickCommand tickCommand)
+    {
+        for (int i = 0; i < WeaponGroupCommands.Length; i++) {
+            if (tickCommand.Has(WeaponGroupCommands[i]))
+                return WeaponGroupCommands[i];
+        }
+        return TickCommands.None;
+    }
+
+    private ConfigWeaponSlots[,] GetWeaponGroups() => new ConfigWeaponSlots[4,3] {
```

Comment:
This will allocate a new array each time it gets executed and will put it the garbage collector.

To get around this we can allocate an empty one as a member variable:
ConfigWeaponSlots[,] m_weaponGroups = new ConfigWeaponSlots[4,3];

Change GetWeaponGroups to populate in the indices and return m_weaponGroups.

Reply:
Did what you asked, although I still populate in the indices every time to make sure that any changes in the config get propagated to the weapon groups

Reply:
Perfect, that is correct 👍